projects/fsastorefrontlib/src/cms-components/agent/appointment-scheduling/appointment-scheduling-form.component.ts

```diff
@@ -0,0 +1,147 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  OnDestroy,
+  OnInit,
+} from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { ActivatedRoute, Params } from '@angular/router';
+import {
+  GlobalMessageService,
+  GlobalMessageType,
+  OCC_USER_ID_ANONYMOUS,
+  RoutingService,
+} from '@spartacus/core';
+import { Subscription } from 'rxjs';
+import { AgentSearchService } from '../../../core/agent/facade/agent-search.service';
+import { UserAccountFacade } from '@spartacus/user/account/root';
+import { DateConfig } from './../../../core/date-config/date-config';
+import { AppointmentSchedulingService } from './../../../core/appointment-scheduling/facade/appointment-scheduling.service';
+import { tap } from 'rxjs/operators';
+
+@Component({
+  selector: 'cx-fs-appointment-scheduling-form',
+  templateUrl: './appointment-scheduling-form.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class AppointmentSchedulingFormComponent implements OnInit, OnDestroy {
+  constructor(
+    protected agentSearchService: AgentSearchService,
+    protected userAccountFacade: UserAccountFacade,
+    protected route: ActivatedRoute,
+    protected fb: FormBuilder,
+    protected globalMessageService: GlobalMessageService,
+    protected router: RoutingService,
+    protected appointmentSchedulingService: AppointmentSchedulingService,
+    protected config: DateConfig
+  ) {}
+
+  private subscription = new Subscription();
+
+  agentId: string;
+  userId: string;
+  availableTimes = Array(6)
+    .fill(null)
+    .map((_elem, index) => ({
+      label: `${index + 10}:00 (1h)`,
+      value: `${index + 10}:00`,
+    }));
+  date = new Date();
+
+  form: FormGroup = this.fb.group({
+    subject: ['', [Validators.required]],
+    appointmentDate: [null, [Validators.required]],
+    appointmentTime: ['', [Validators.required]],
+    description: ['', [Validators.required]],
+    consentGiven: [false, [Validators.required, Validators.requiredTrue]],
+  });
+
+  ngOnInit() {
+    this.subscription
+      .add(this.getAgentId())
+      .add(this.getUserId())
+      .add(this.onConsentChange());
+  }
+
+  goBack() {
+    this.router.back();
+  }
+
+  private getAgentId(): Subscription {
+    return this.route.params.subscribe((params: Params) => {
+      if (params) {
+        this.agentId = params.agent;
+      }
+    });
+  }
+
+  private getUserId(): Subscription {
+    return this.userAccountFacade.get().subscribe(user => {
+      user && user.uid
+        ? (this.userId = user.uid)
+        : (this.userId = OCC_USER_ID_ANONYMOUS);
+    });
+  }
+
+  private onConsentChange(): Subscription {
+    return this.form.get('consentGiven').valueChanges.subscribe(val => {
+      if (val) {
+        this.globalMessageService.add(
+          'Consent is successfully given.',
```

Comment:
Do not hard code message, use localization instead. Define new key in translation files, for example appointmentScheduling.successfullConsent

Reply:
```
            this.globalMessageService.add(
              {
                key: 'appointmentScheduling.successfullConsent',
              },
              GlobalMessageType.MSG_TYPE_CONFIRMATION
            );
```

Also, the third property is not needed, i guess that it is indicator how many seconds will this global message be displayed. It is controlled in FSA on the configuration level. (check defaultFSGlobalMessageConfigFactory)